api/controllers/user.go

```diff
@@ -0,0 +1,38 @@
+package controllers
+
+import (
+  "net/http"
+  "strings"
+  "github.com/gin-gonic/gin"
+  "github.com/charmbracelet/log"
+
+  "github.com/Salz-Team/salz/api/db"
+)
+
+func GetUser(c *gin.Context, dbHandler db.ApiDBHandler, authHandler db.AuthDBHandler) {
+  // Get the user from the database
+  
+  authTokenHeader := c.GetHeader("Authorization")
+  authToken := strings.TrimSpace(strings.Replace(authTokenHeader, "Bearer", "", 1))
+
+  token, err := authHandler.GetToken(authToken)
+  if err != nil {
+    log.Error("Unable to get token", "error", err)
+    c.AbortWithStatus(http.StatusInternalServerError)
+    return
+  }
+
+  user, err := dbHandler.GetUser(token.UserId)
+  if err != nil {
+    log.Error("Unable to get user", "error", err)
+    c.AbortWithStatus(http.StatusInternalServerError)
+    return
+  }
```

Comment:
If the row doesn't exist, maybe return 401? FE can then drop the token, which logs them out, and then redirect user back to homepage, so that they would have to log in again.

Reply:
Sure. I still think its an internal error and that 500s are more likely to be investigated by us than 401s, but perhaps a scarier log message would do.